src/controllers/request-controller.ts

```diff
@@ -73,22 +83,25 @@ export class RequestController {
       }
       const streamId = StreamID.fromString(req.body.streamId)
 
+      // Store metadata from genesis to the database
+      await this.metadataService.fill(streamId)
```

Comment:
If not this line, CAS would not have metadata on the database at all. Now, it would not break anything. After Merkle construction is changed, absence of the line would break everything.

Reply:
Thanks! Do you plan to encapsulate all of the calls needed for the new Merkle tree construction to work in a single interface/method? If so, would it make sense to add a TODO here?